xbmc/platform/linux/powermanagement/LogindUPowerSyscall.cpp

```diff
@@ -187,154 +171,83 @@ int CLogindUPowerSyscall::BatteryLevel()
-    bool isRechargeable = properties["IsRechargeable"].asBoolean();
+    auto device = sdbus::createProxy("org.freedesktop.UPower", i);
+    auto rechargable =
+        device->getProperty("IsRechargeable").onInterface("org.freedesktop.UPower.Device");
+    auto percentage =
+        device->getProperty("Percentage").onInterface("org.freedesktop.UPower.Device");
 
-    if (isRechargeable)
+    if (rechargable)
     {
       batteryCount++;
-      batteryLevelSum += properties["Percentage"].asDouble();
+      batteryLevelSum += percentage.get<double>();
     }
   }
 
-  dbus_free_string_array(source);
-
   if (batteryCount > 0)
   {
-    m_batteryLevel = (int)(batteryLevelSum / (double)batteryCount);
-    m_lowBattery = CDBusUtil::GetVariant("org.freedesktop.UPower", "/org/freedesktop/UPower", "org.freedesktop.UPower", "OnLowBattery").asBoolean();
+    m_batteryLevel = static_cast<int>(batteryLevelSum / batteryCount);
   }
 }
 
 bool CLogindUPowerSyscall::PumpPowerEvents(IPowerEventsCallback *callback)
 {
-  bool result = false;
-  bool releaseLockSleep = false;
-
-  if (m_connection)
+  if (m_OnSuspend)
   {
-    dbus_connection_read_write(m_connection, 0);
-    DBusMessagePtr msg(dbus_connection_pop_message(m_connection));
-
-    if (msg)
-    {
-      if (dbus_message_is_signal(msg.get(), "org.freedesktop.login1.Manager", "PrepareForSleep"))
-      {
-        dbus_bool_t arg;
-        // the boolean argument defines whether we are going to sleep (true) or just woke up (false)
-        dbus_message_get_args(msg.get(), NULL, DBUS_TYPE_BOOLEAN, &arg, DBUS_TYPE_INVALID);
-        CLog::Log(LOGDEBUG, "LogindUPowerSyscall: Received PrepareForSleep with arg %i", (int)arg);
-        if (arg)
-        {
-          callback->OnSleep();
-          releaseLockSleep = true;
-        }
-        else
-        {
-          callback->OnWake();
-          InhibitDelayLockSleep();
-        }
-
-        result = true;
-      }
-      else if (dbus_message_is_signal(msg.get(), "org.freedesktop.UPower", "DeviceChanged"))
-      {
-        bool lowBattery = m_lowBattery;
-        UpdateBatteryLevel();
-        if (m_lowBattery && !lowBattery)
-          callback->OnLowBattery();
-
-        result = true;
-      }
-      else
-        CLog::Log(LOGDEBUG, "LogindUPowerSyscall - Received unknown signal %s", dbus_message_get_member(msg.get()));
-    }
+    callback->OnSleep();
+    m_OnSuspend = false;
+    m_OnResume = true;
+    return true;
+  }
+  else if (m_OnResume)
+  {
+    callback->OnWake();
+    m_OnResume = false;
+    return true;
   }
 
-  if (releaseLockSleep)
-    ReleaseDelayLockSleep();
-
-  return result;
+  return false;
 }
 
 void CLogindUPowerSyscall::InhibitDelayLockSleep()
 {
-  m_delayLockSleepFd = InhibitDelayLock("sleep");
+  m_proxy->callMethod("Inhibit")
+      .onInterface("org.freedesktop.login1.Manager")
+      .withArguments("sleep", "Kodi", "", "delay")
```

Comment:
There a way not to use Kodi here seeing as you have another PR open to get rid of it hardcoding?